In musl open64 is by default a #define to open.
However, it provides compat symbols and we want this
to explicitly use those.

--- a/libeatmydata/libeatmydata.c
+++ b/libeatmydata/libeatmydata.c
@@ -31,6 +31,12 @@
 #include <pthread.h>
 #endif
 
+/* refer to the compat alias in musl */
+#if !defined(__GLIBC__)
+#undef open64
+typedef off_t off64_t;
+#endif
+
 /* 
 #define CHECK_FILE "/tmp/eatmydata"
 */
